### ISSUE TITLE
Add left/right direction to tooltipIcon

### ISSUE THREAD

**Issue by filipmadej** [carbon-design-system/carbon-components-react#1306]
### Summary

I would like to have an option left/right added for TooltipIcon

### Justification

I have a problem with TooltipIcon as it’s getting cut by the window. I could solve it by moving tooltip to the left but supported direction are only top and bottom. Regular tooltips support left/right.

![screen shot 2018-09-10 at 16 33 37](https://user-images.githubusercontent.com/19649061/45433235-00de5100-b6ac-11e8-8b6a-77799d156d0c.png)

**Comment by asudoh** [comment]
@IBM/carbon-designers I believe we had a conscious decision to make it top/bottom only, but please let us know if I’m wrong here - Thanks!

**Comment by filipmadej** [comment]
If that's the case then can we add some logic to move tooltip to viewable area like so:
![screen shot 2018-09-13 at 13 40 40](https://user-images.githubusercontent.com/19649061/45486226-a64ffe00-b75a-11e8-8285-c9693183e51c.png)

**Comment by aagonzales** [comment]
Yes, only top and bottom because opening left or right could cover up associated information. However, moving the sharkfin to the left or right side is acceptable. All of the below are correct. With the icon-tooltip it should be 8px from either edge.

![image](https://user-images.githubusercontent.com/11670886/45685125-41245000-bb40-11e8-9b05-956df19278f2.png)

**Comment by asudoh** [comment]
@filipmadej Given the suggestion from @aagonzales, we'll monitor how many people would like this feature. Meanwhile, you may want to try something like applying `right:0` to `bx--tooltip--definition__bottom` and `left: auto`/`margin-right: 0.5rem` to `.bx--tooltip__caret`.

However, @aagonzales' above words seems to imply that definition tooltip for explaining words, which may conflict the icon usage shown by @filipmadej.

**Comment by asudoh** [comment]
~~@filipmadej OK we thought it's definition tooltip given your screenshot. Anyhow icon-based tooltip has a different story - `<Tooltip>` has [`menuOffset` prop](https://github.com/IBM/carbon-components-react/blob/v6.32.2/src/components/Tooltip/Tooltip.js#L140-L146), which takes an object with `top`/`left` properties or a function returning that object. You can adjust the position by that. You can adjust the position of caret by CSS. Hope it helps!~~

**Comment by aagonzales** [comment]
I think the rule would apply to all 3 kinds of tooltips.

**Comment by filipmadej** [comment]
@asudoh looking at the code of ```<TooltipIcon>``` I don't see menuOffset prop. Is someone working on adding such a prop to tooltip icon? Because I don't see how you answer might help in my situation if prop doesn't exists.

**Comment by asudoh** [comment]
`<TooltipIcon>` takes a route of style adjustment. You can do it youself by editing the style in DOM inspector. Below is something you may end up with:

```css
.your--tooltip-icon__bottom.bx--tooltip--icon__bottom:after {
  transform: translate(0%, calc(100% + 10px));
  margin-left: 0;
  right: 0;
}
```
